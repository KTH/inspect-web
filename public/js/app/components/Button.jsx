import React, { useState } from 'react'

function Button({ caption = 'N/A' }) {
  const [buttonClicked, setButtonClicked] = useState(false)

  return (
    <>
      <button type="button" onClick={() => setButtonClicked(true)}>
        {caption}
      </button>
      <p data-testid="click-message">{buttonClicked ? 'Button works!' : ''}</p>
    </>
  )
}

export default Button
