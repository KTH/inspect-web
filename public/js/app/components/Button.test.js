import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from './Button'

test('loads and displays greeting', async () => {
  const { asFragment } = render(<Button message="This is a message" />)
  expect(asFragment()).toMatchSnapshot()

  expect(screen.getByTestId('message-header')).toHaveTextContent('This is a message')

  fireEvent.click(screen.getByText('Try me'))
  expect(screen.getByTestId('click-message')).toHaveTextContent('Button works!')
  expect(asFragment()).toMatchSnapshot()
})
