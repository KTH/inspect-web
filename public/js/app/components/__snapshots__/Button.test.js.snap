// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greeting 1`] = `
<DocumentFragment>
  <h1
    class="display-3"
  >
    Node-web
  </h1>
  <h2>
    You are upp and running kth-node react!
  </h2>
  <hr
    class="my-2"
  />
  <h3
    data-testid="message-header"
  >
    Message from routerStore: This is a message
  </h3>
  <button
    type="button"
  >
    Try me
  </button>
  <p
    data-testid="click-message"
  />
</DocumentFragment>
`;

exports[`loads and displays greeting 2`] = `
<DocumentFragment>
  <h1
    class="display-3"
  >
    Node-web
  </h1>
  <h2>
    You are upp and running kth-node react!
  </h2>
  <hr
    class="my-2"
  />
  <h3
    data-testid="message-header"
  >
    Message from routerStore: This is a message
  </h3>
  <button
    type="button"
  >
    Try me
  </button>
  <p
    data-testid="click-message"
  >
    Button works!
  </p>
</DocumentFragment>
`;
